lib/travis/api/v3/queries/repositories.rb

```diff
@@ -11,6 +11,7 @@ def all
         all = Models::Repository
         all = all.where(active:  bool(active))  unless active.nil?
         all = all.where(private: bool(private)) unless private.nil?
+        all = all.includes(:default_branch) # TODO: use includes params
```

Comment:
This is where we actually avoid the N+1 query. That one line is what all the other fun was for.